geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePidTest.java

```diff
@@ -100,7 +105,53 @@ public void findAvailablePidsShouldReturnNoDuplicatedPids() throws Exception {
     assertThatNoPidIsDuplicated(availablePid.findAvailablePids(8));
   }
 
-  private void assertThatNoPidIsDuplicated(int[] pids) {
+  @Test(timeout = DEFAULT_TIMEOUT_MILLIS)
+  public void findAvailablePidShouldReturnGreaterThanOrEqualToLowerBound() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    int pid = availablePid.findAvailablePid();
+
+    assertThat(pid).isGreaterThanOrEqualTo(1);
+  }
+
+  @Test(timeout = DEFAULT_TIMEOUT_MILLIS)
+  public void findAvailablePidShouldReturnLessThanOrEqualToUpperBound() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    int pid = availablePid.findAvailablePid();
+
+    assertThat(pid).isLessThanOrEqualTo(3);
+  }
+
+  @Test
+  public void randomLowerBoundIsInclusive() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    await().atMost(10, SECONDS).until(() -> assertThat(availablePid.random()).isEqualTo(1));
+  }
+
+  @Test
+  public void randomUpperBoundIsInclusive() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    await().atMost(10, SECONDS).until(() -> assertThat(availablePid.random()).isEqualTo(3));
```

Comment:
This is a clever way to test that lower and upper bounds are potential output values!